### PR TITLE
[Filesystem] Get the file or directory permissions

### PR DESCRIPTION
**Pull request by SofHad**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | no |
| License | MIT |
| Doc PR | no |

``` php
// Use
$fs->permissions($dirOrFile);         //0755
$fs->permissions($dirOrFile, true);   //u-wxrw--wt
```

### CODE REVIEW

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 1]
```diff
@@ -541,6 +541,67 @@ public function dumpFile($filename, $content, $mode = 0666)
...
+     *                        - => regular;
+     *                        b => block special;
+     *                        d => directory;
+     *                        c => character special;
+     *                        p => FIFO pipe;
+     *                        u => unknown.
+     *
+     * @return string Permissions of the file|directory or throw an exception on failure.
+     */
+    public function permssions($path, $groups = false)
```
Comment (lyrixx) [inline src/Symfony/Component/Filesystem/Filesystem.php]: You forgot the I in permissions

Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: done :)

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 2]
```diff
@@ -541,6 +541,67 @@ public function dumpFile($filename, $content, $mode = 0666)
...
+     *                        b => block special;
+     *                        d => directory;
+     *                        c => character special;
+     *                        p => FIFO pipe;
+     *                        u => unknown.
+     *
+     * @return string Permissions of the file|directory or throw an exception on failure.
+     */
+    public function permissions($path, $groups = false)
+    {
```
Comment (iltar) [inline src/Symfony/Component/Filesystem/Filesystem.php]: Personally I would prefer to have a constant defining the flags of groups.  

What about calling it `getPremissions()` instead? That would be a lot easier to use as you expect to get something thus start guessing the method name like that.

File: src/Symfony/Component/Filesystem/Filesystem.php [hunk 3]
```diff
@@ -541,6 +541,67 @@ public function dumpFile($filename, $content, $mode = 0666)
...
+            $out = 'l';
+        } elseif (($perms & 0x8000) == 0x8000) {
+            $out = '-';
+        } elseif (($perms & 0x6000) == 0x6000) {
+            $out = 'b';
+        } elseif (($perms & 0x4000) == 0x4000) {
+            $out = 'd';
+        } elseif (($perms & 0x2000) == 0x2000) {
+            $out = 'c';
+        } elseif (($perms & 0x1000) == 0x1000) {
```
Comment (iltar) [inline src/Symfony/Component/Filesystem/Filesystem.php]: Would it be an idea to move all those hexes to constants so everyone knows what they mean?

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 2 above)
Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: I like this name *getPremissions*

Comment (javiereguiluz) [inline src/Symfony/Component/Filesystem/Filesystem.php]: In the `Filesystem` class we always mimic the names of the equivalent PHP functions. So this new function "must" be called [fileperms](http://php.net/fileperms) instead. Using the same names reduces the learning curve.

Comment (iltar) [inline src/Symfony/Component/Filesystem/Filesystem.php]: @javiereguiluz fair enough

File: src/Symfony/Component/Filesystem/Filesystem.php (hunk 3 above)
Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: it makes a lot of constants imo ?

Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: approximately 30 constants :)

Comment (Arthurhall) [inline src/Symfony/Component/Filesystem/Filesystem.php]: You can put all these constants in a static Class named "Permissions" ?
-> Symfony/Component/Filesystem/Permissions

Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: +1 for *Symfony/Component/Filesystem/Permissions*

Comment (javiereguiluz) [inline src/Symfony/Component/Filesystem/Filesystem.php]: Before doing all these changes ... I'd wait for the approval of this feature by the Symfony Core members.

Comment (SofHad) [inline src/Symfony/Component/Filesystem/Filesystem.php]: @javiereguiluz ok

**Comment by fabpot** [comment]
What is the use case? The Filesystem class mainly exists to abstract built-in PHP functions. I'm :-1: to add this.

**Pull request closed without merging**